### PR TITLE
Add LowerLayerSetter class to allow to tweak coap level parameter for requests

### PR DESCRIPTION
**Pull request by sbernard31**
This aims to implements #875.

An example of use : 
```java
// configure reliability parameters
Builder reliabilityParams = ReliabilityLayerParameters.builder();
reliabilityParams.ackRandomFactor(1).ackTimeout(100).ackTimeoutScale(1).maxRetransmit(1);
// send requests
ReadResponse response = server.send(registration,
                                    new ReadRequest("3/0/1"),  
                                    CoapRequestSetter.reliabilitySetter(reliabilityParams), 
                                    timeout);
```

### CODE REVIEW

**Comment by sbernard31** [comment]
The master version (without any backward compatibility stuff and with support for bootstrap server is available at https://github.com/eclipse/leshan/commit/7a2966a3c59b89db1aa832d42ab7191afb9e6ff6)

File: leshan-server-core/src/main/java/org/eclipse/leshan/server/request/LowerLayerSetter.java [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+package org.eclipse.leshan.server.request;
+
+/**
+ * Allow to apply custom setting to lower layer.
+ * <p>
+ * If you are using LWM2M over CoAP this setter will help you to apply CoAP setting to you request.
+ * 
+ * @since 1.2
+ */
+public interface LowerLayerSetter {
```
Comment (msangoi) [inline leshan-server-core/src/main/java/org/eclipse/leshan/server/request/LowerLayerSetter.java:25]: WDYT of `LowerLayerConfig`? with an `apply()` method

Comment (sbernard31) [reply leshan-server-core/src/main/java/org/eclipse/leshan/server/request/LowerLayerSetter.java:25]: :+1:  sounds good. I will modify it.

**Comment by sbernard31** [comment]
This now integrated in branch 1.x (cf7dbc9 ) with backward compatibility
and in master commit (813d6642b) without backward compatibility.

**Pull request merged**